### ISSUE TITLE
ResNeSt Cascade RCNN S101 doesn't support mini-batch size of 1

### ISSUE THREAD

**Issue by waddington** [open-mmlab/mmdetection#5098]
1. Use the ResNeSt Cascade RCNN S101 config
2. Change the `samples_per_gpu` property to 1.

Result: an error that the number of channels for the input to the model is incorrect.

I think that MMDetection should allow mini-batch size of 1 here to cater for people that cannot fit multiple samples onto their GPU. With the default config, I expect a mini-batch size of 1 will save a couple of GB compared to size of 2.

**Comment by ZwwWayne** [comment]
Seems to be a bug. Please @AronLin take a look.

**Comment by AronLin** [comment]
Maybe you have not set model.train() or model.eval()?
In mmdet, model.train() or model.eval() will set our `BatchNorm` to evaluation mode.
Or the BN will verify the input size, when the input size is 1, there will be an assert error.
